jmix-flowui/flowui-kit/src/main/java/io/jmix/flowui/kit/component/grid/GridMenuItemActionSupport.java

```diff
@@ -42,32 +48,72 @@ public Action getAction() {
     }
 
     public void setAction(@Nullable Action action) {
-        if (Objects.equals(this.action, action)) {
+        setAction(action, true);
+    }
+
+    public void setAction(@Nullable Action action, boolean overrideComponentProperties) {
+        if (Objects.equals(this.action, action) && this.overrideComponentProperties == overrideComponentProperties) {
             return;
         }
 
         removeRegistrations();
 
         this.action = action;
+        this.overrideComponentProperties = overrideComponentProperties;
 
         if (action != null) {
-            menuItem.setText(action.getText());
-            menuItem.setEnabled(action.isEnabled());
-            menuItem.setVisible(action.isVisible());
-
-            registration = menuItem.addMenuItemClickListener(event ->
+            GridMenuItem<?> contextMenuItem = menuItem.getMenuItem();
+            if (overrideComponentProperties) {
+                menuItem.setText(action.getText());
+                contextMenuItem.setEnabled(action.isEnabled());
+                contextMenuItem.setVisible(action.isVisible());
+                menuItem.setTooltipText(action.getDescription());
+                if (isShowActionIconEnabled()) {
+                    menuItem.setPrefixComponent(createIconComponent(action.getIcon()));
+                }
+                if (isShowActionShortcutEnabled()) {
+                    menuItem.setSuffixComponent(createShortcutComponent(action.getShortcutCombination()));
+                }
+                actionPropertyChangeRegistration = addPropertyChangeListener();
+            }
+            registration = contextMenuItem.addMenuItemClickListener(event ->
                     action.actionPerform(event.getSource()));
-            actionPropertyChangeRegistration = addPropertyChangeListener();
         }
 
         updateVisible();
     }
 
+    protected boolean isShowActionIconEnabled() {
+        return false;
+    }
+
+    @Nullable
+    protected Component createIconComponent(@Nullable Icon actionIcon) {
+        if (actionIcon == null) {
+            return null;
+        }
+        String iconAttribute = actionIcon.getElement().getAttribute("icon");
+        if (iconAttribute == null) {
+            return null;
+        }
+        return ComponentUtils.parseIcon(iconAttribute);
+    }
+
+    protected boolean isShowActionShortcutEnabled() {
+        return false;
+    }
+
+    @Nullable
+    protected Component createShortcutComponent(@Nullable KeyCombination keyCombination) {
+        return keyCombination == null ? null : new Span(keyCombination.format());
```

Comment:
let's replace span with `new Html("<kbd>" + keyCombination.format() + "</kbd>")`

also we need to style `<kbd>`, e.g.: something like this:

```
.jmix-grid-context-menu-item-component kbd {
    box-sizing: border-box;
    display: inline-block;
    padding: calc(var(--lumo-space-xs) / 2) var(--lumo-space-xs);
    font-size: calc(var(--lumo-font-size-xs) * 0.85);
    line-height: var(--lumo-line-height-xs);
    color: var(--lumo-secondary-text-color);
    vertical-align: middle;
    background-color: var(--lumo-contrast-5pct);
    border: 1px solid var(--lumo-contrast-20pct);
    border-radius: var(--lumo-border-radius-m);
    box-shadow: inset 0 -1px 0 var(--lumo-contrast-20pct, var(--lumo-contrast-5pct));
}
```

result: 

<img width="207" alt="Screenshot 2024-01-18 at 20 19 36" src="https://github.com/jmix-framework/jmix/assets/6077149/a3f20e23-bb39-450d-a5ef-43e8ce11c1e4">